extendr-api/src/robj.rs

```diff
@@ -134,6 +134,26 @@ impl<'a> FromRobj<'a> for Vec<f64> {
     }
 }
 
+macro_rules! impl_iter_from_robj {
+    ($t: ty, $iter_fn: ident) => {
+        impl<'a> FromRobj<'a> for $t {
+            fn from_robj(robj: &'a Robj) -> Result<Self, &'static str> {
+                if let Some(v) = robj.$iter_fn() {
+                    Ok(v)
+                } else {
+                    Err("not a vector of strings")
```

Comment:
Is this error message correct? Shouldn't it depend on the type for which you're generating an iterator?